src/flightctl/kubeconfig_writer.rs

```diff
@@ -0,0 +1,81 @@
+use super::kubectl;
+use kube::config::{NamedAuthInfo, NamedCluster};
+use std::io::Write;
+use tempfile::NamedTempFile;
+
+pub fn write_auth(auth: &NamedAuthInfo) -> anyhow::Result<()> {
+    let mut args = vec![
+        String::from("config"),
+        String::from("set-credentials"),
+        auth.name.clone(),
+    ];
+    if let Some(exec) = &auth.auth_info.exec {
+        args.push(String::from("--exec-command"));
+        args.push(exec.command.clone());
+
+        if let Some(api_version) = &exec.api_version {
+            args.push(String::from("--exec-api-version"));
+            args.push(api_version.clone())
+        }
+
+        if let Some(exec_args) = &exec.args {
+            for arg in exec_args {
+                args.push(String::from("--exec-arg"));
+                args.push(arg.clone());
```

Comment:
To answer your question from Slack, I don't *think* there'd be a reasonable way to get rid of these `clone()`s

Reply:
I agree, so long as you are taking references. But since `write_auth` and `ensure_auth` is called exactly once, you could take ownership of it instead.

Reply:
@mike-burns Good point; I was able to remove `clone` from `kubeconfig_writer` by moving instead of borrowing (4bb323a). There are some shared references I still had to clone in `context`, at least for now.